test/app-luatest/gh_7746_fiber_channel_close_test.lua

```diff
@@ -0,0 +1,55 @@
+local compat = require('tarantool').compat
+local t = require('luatest')
+local g = t.group()
+local fiber = require('fiber')
+
+g.test_channel_close_default = function()
+    -- No graceful close (old) by default.
+    local ch = fiber.channel(10)
+    ch:put('one')
+    ch:put('two')
+    t.assert_equals(ch:get(), 'one')
+    ch:close()
+    t.assert(ch:is_closed())
+    t.assert_not(ch:put('three'))
+    t.assert_equals(ch:get(), nil)
+end
+
+g.test_channel_close_old = function()
+    compat.fiber_channel_graceful_close = 'old'
+    local ch = fiber.channel(10)
+    ch:put('one')
+    ch:put('two')
+    t.assert_equals(ch:get(), 'one')
+    ch:close()
+    t.assert(ch:is_closed())
+    t.assert_not(ch:put('three'))
+    t.assert_equals(ch:get(), nil)
+end
+
+g.test_channel_close_new = function()
+    -- Graceful close (new) selected.
+    compat.fiber_channel_graceful_close = 'new'
+    local ch = fiber.channel(10)
+    ch:put('one')
+    ch:put('two')
+    ch:close()
+
+    t.assert(ch:is_closed())
+    t.assert_not(ch:put('three'))
+
+    t.assert_equals(ch:get(), 'one')
+    t.assert(ch:is_closed())
+
+    t.assert_equals(ch:get(), 'two')
+    t.assert(ch:is_closed())
```

Comment:
We can do one more `:get()` to ensure that it'll not wait for writers.

Feel free to ignore.

Reply:
Good idea, done.